packages/expo-battery/android/src/main/java/expo/modules/battery/BatteryStateReceiver.kt

```diff
@@ -0,0 +1,34 @@
+package expo.modules.battery
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.BatteryManager
+import android.os.Bundle
+import org.unimodules.core.interfaces.services.EventEmitter
+
+class BatteryStateReceiver(private val eventEmitter: EventEmitter?) : BroadcastReceiver() {
+  private val BATTERY_CHARGED_EVENT_NAME = "Expo.batteryStateDidChange"
+
+  fun batteryStatusNativeToJS(status: Int): BatteryModule.BatteryState {
+    return when (status) {
+      BatteryManager.BATTERY_STATUS_FULL -> BatteryModule.BatteryState.FULL
+      BatteryManager.BATTERY_STATUS_CHARGING -> BatteryModule.BatteryState.CHARGING
+      BatteryManager.BATTERY_STATUS_NOT_CHARGING -> BatteryModule.BatteryState.UNPLUGGED
+      BatteryManager.BATTERY_STATUS_DISCHARGING -> BatteryModule.BatteryState.UNPLUGGED
+      else -> BatteryModule.BatteryState.UNKNOWN
+    }
+  }
```

Comment:
We have the same function in `BatteryModule.kt`. Please, extract it into a separate file. We don't want to have the same code in multiple places.